Sources/MapboxNavigation/StepsViewController.swift

```diff
@@ -192,18 +192,24 @@ public class StepsViewController: UIViewController {
 extension StepsViewController: UITableViewDelegate {
     public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
+        let legIndex = indexPath.section
         let cell = tableView.cellForRow(at: indexPath) as! StepTableViewCell
         // Since as we progress, steps are removed from the list, we need to map the row the user tapped to the actual step on the leg.
         // If the user selects a step on future leg, all steps are going to be there.
         var stepIndex: Int
-        if indexPath.section > 0 {
```

Comment:
Do you think that it'd be possible to extract this logic into separate function and add unit-test for it? With unit-test we can easily provide negative test case scenario, which will help in future.

Reply:
Thanks for the suggestion! I'll add test for this part.